infrastructure/templates/public-api/components/functionApp.bicep

```diff
@@ -143,14 +169,47 @@ resource functionApp 'Microsoft.Web/sites@2023-01-01' = {
       preWarmedInstanceCount: preWarmedInstanceCount ?? null
       netFrameworkVersion: '8.0'
       linuxFxVersion: appServicePlanOS == 'Linux' ? 'DOTNET-ISOLATED|8.0' : null
+      keyVaultReferenceIdentity: userAssignedManagedIdentityParams != null ? userAssignedManagedIdentityParams!.id : null
     }
+    keyVaultReferenceIdentity: userAssignedManagedIdentityParams != null ? userAssignedManagedIdentityParams!.id : null
+    publicNetworkAccess: publicNetworkAccessEnabled ? 'Enabled' : 'Disabled'
   }
   tags: tagValues
 }
 
+resource azureStorageAccounts 'Microsoft.Web/sites/config@2021-01-15' = {
+   name: 'azurestorageaccounts'
+   parent: functionApp
+   properties: reduce(azureFileShares, {}, (cur, next) => union(cur, {
+     '${next.storageName}': {
+       type: 'AzureFiles'
+       shareName: next.fileShareName
+       mountPath: next.mountPath
+       accountName: next.storageAccountName
+       accessKey: next.storageAccountKey
+     }
+   }))
+}
+
+resource slot1StorageAccountFileShare 'Microsoft.Storage/storageAccounts/fileServices/shares@2023-01-01' = {
+  name: '${durableManagementStorageAccountName}/default/${fullFunctionAppName}1'
```

Comment:
I don't understand why the name here is `durableManagementStorageAccountName` rather than `slot1StorageAccountName`?

Reply:
This was by design, opting to put the file shares that are used for the individual slots to hold their code into the shared storage account.

However, I'm happy to change this to try to put it into the dedicated storage accounts for slot1 and slot2 though and see how it goes.

Reply:
Seems to work fine and makes a lot more sense - I've made the change thanks.